src/code/client.ts

```diff
@@ -645,6 +663,8 @@ class CloudFileManagerClient {
             return this.alert(err)
           }
         }
+        metadata.autoSaveDisabled = true;
+        this._setState({ metadata, saving: null })
```

Comment:
I think this warrants a comment. The effect seems to be that we disable autosave after the second non-40x failure but not the first, but I'm not sure I understand the rationale.

Reply:
Thanks for spotting this! That was definitely not my intention. Clearly this change was poorly thought out. Will move these lines to where they will have better effect.